tests/phpunit/tests/query/cacheResults.php

```diff
@@ -1418,4 +1418,66 @@ public function test_get_post_meta_lazy_loads_all_term_meta_data() {
 		// No additional queries expected.
 		$this->assertSame( 0, $num_queries, 'Unexpected number of queries during second query of term meta.' );
 	}
+
+	/**
+	 * @ticket 59442
+	 *
+	 * @covers WP_Query::generate_cache_key
+	 * @dataProvider data_duplicate_query_when_sticky_post
+	 */
+	public function test_duplicate_query_when_sticky_post( $is_cached_prior, $expected_num_queries ) {
```

Comment:
I tried running these tests in trunk without the rest of the changes applied and they're already passing, which makes me think that this isn't an effective test.

Reply:
This case should pass before and after. That is the point, to show the behaviour didn't change. This test just covers a code path that was not covered before.

Reply:
Thanks, Jonny. In that case, I think we need another test case that demonstrates the bug and confirms that this change fixes the issue.